numpy/testing/_private/utils.py

```diff
@@ -769,16 +769,19 @@ def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
         y_id = func(y)
         # We include work-arounds here to handle three types of slightly
         # pathological ndarray subclasses:
-        # (1) all() on `masked` array scalars can return masked arrays, so we
-        #     use != True
+        # (1) all() on fully masked arrays returns np.ma.masked, so we use != True
+        #     (np.ma.masked != True evaluates as np.ma.masked, which is falsy).
         # (2) __eq__ on some ndarray subclasses returns Python booleans
-        #     instead of element-wise comparisons, so we cast to np.bool() and
-        #     use isinstance(..., bool) checks
+        #     instead of element-wise comparisons, so we cast to np.bool in
+        #     that case.
         # (3) subclasses with bare-bones __array_function__ implementations may
         #     not implement np.all(), so favor using the .all() method
-        # We are not committed to supporting such subclasses, but it's nice to
+        # We are not committed to supporting cases (2) and (3), but it's nice to
         # support them if possible.
-        if np.bool(x_id == y_id).all() != True:
+        result = x_id == y_id
+        if isinstance(result, bool):
```

Comment:
This feels pretty brittle to me. Is there another way to do this without `isinstance`?

Reply:
Pushed a more robust approach